src/test/java/com/redhat/devtools/lsp4ij/features/formatting/TypeScriptEditorImprovementsTest.java

```diff
@@ -0,0 +1,509 @@
+import com.redhat.devtools.lsp4ij.server.definition.launching.ClientConfigurationSettings;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Tests for LSPEditorImprovementsTypedHandler, LSPEditorImprovementsBackspaceHandler, and LSPEditorImprovementsEnterBetweenBracesHandler.
+ */
+public class TypeScriptEditorImprovementsTest extends LSPCodeInsightFixtureTestCase {
+
+    private static final String TEST_FILE_NAME = "test.ts";
+    private static final char BACKSLASH = '\\';
+    private static final String CARET = "<caret>";
+
+    public TypeScriptEditorImprovementsTest() {
+        super("*.ts");
+        setClientConfigurable(true);
+    }
+
+    @Override
+    protected void tearDown() throws Exception {
+        // Restore the default code style
+        Editor editor = myFixture.getEditor();
+        if (editor != null) {
+            CodeStyle.getSettings(editor).getIndentOptions().USE_TAB_CHARACTER = false;
+        }
+
+        super.tearDown();
+    }
+
+    @NotNull
+    private LanguageServerItem initializeLanguageServer() {
+        List<LanguageServerItem> languageServers = new LinkedList<>();
+        try {
+            Project project = myFixture.getProject();
+            PsiFile file = myFixture.getFile();
+            ContainerUtil.addAllNotNull(languageServers, LanguageServiceAccessor.getInstance(project)
+                    .getLanguageServers(file.getVirtualFile(), null, null)
+                    .get(5000, TimeUnit.MILLISECONDS));
+        } catch (Exception e) {
+            fail(e.getMessage());
+        }
+        LanguageServerItem languageServer = ContainerUtil.getFirstItem(languageServers);
+        assertNotNull(languageServer);
+        return languageServer;
+    }
+
+    @NotNull
+    private ClientConfigurationSettings getClientConfigurationSettings(@NotNull LanguageServerItem languageServer) {
+        LanguageServerDefinition languageServerDefinition = languageServer.getServerDefinition();
+        assertInstanceOf(languageServerDefinition, ClientConfigurableLanguageServerDefinition.class);
+        ClientConfigurableLanguageServerDefinition configurableLanguageServerDefinition = (ClientConfigurableLanguageServerDefinition) languageServerDefinition;
+        ClientConfigurationSettings clientConfigurationSettings = configurableLanguageServerDefinition.getLanguageServerClientConfiguration();
+        assertNotNull(clientConfigurationSettings);
+        return clientConfigurationSettings;
+    }
+
+    // These tests exercise both LSPEditorImprovementsTypedHandler.handleNestedQuote() and LSPEditorImprovementsBackspaceHandler
+
+    public void testNestedQuotesInSingleQuotedStringLiteral() {
+        testNestedQuotes('\'', '"');
+        testNestedQuotes('\'', '`');
+    }
+
+    public void testNestedQuotesInSingleQuotedStringLiteralDisabled() {
```

Comment:
There are tests for when the respective editor behavior feature is enabled (default) and disabled.